src/main/java/org/folio/dcb/service/impl/LendingLibraryServiceImpl.java

```diff
@@ -65,6 +66,10 @@ public void updateTransactionStatus(TransactionEntity dcbTransaction, Transactio
       updateTransactionEntity(dcbTransaction, requestedStatus);
     } else if (ITEM_CHECKED_OUT == currentStatus && ITEM_CHECKED_IN == requestedStatus) {
       updateTransactionEntity(dcbTransaction, requestedStatus);
+    } else if(CANCELLED == requestedStatus) {
```

Comment:
@MagzhanArtykov  - Where is the code or where will the check happen for the below mentioned lines ? 
The cancellation request can be applied to DCB transactions that are in the following statuses [CREATED, OPEN, AWAITING_PICKUP]. For the transactions that are in status [ITEM_CHECKED_OUT, ITEM_CHECKED_IN] FOLIO should reject such requests with a "400 Bad request" response.

Reply:
The check will happen on Front-End. Therefore, resolving this conversation.